### ISSUE TITLE
List.delete() sets element to empty array instead of removing it

### ISSUE THREAD

**Issue by lpatalas** [immutable-js/immutable-js#1474]
Hello.

I'm using Immutable.js v4.0.0-rc.9 with react-native v0.50.4.

If I run the following code snippet:
```
const list1 = List.of(1, 2, 3);
const list2 = list1.delete(1);
console.log(`list1`, list1.size, list1);
console.log(`list2`, list2.size, list2);
```

I get this output in the console:

```
01-14 10:22:26.568  4347  4982 I ReactNativeJS: 'list1', 3, { size: 3,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _origin: 0,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _capacity: 3,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _level: 5,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _root: null,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _tail: { array: [ 1, 2, 3 ], ownerID: undefined },
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __ownerID: undefined,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __hash: undefined,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __altered: false }
01-14 10:22:26.568  4347  4982 I ReactNativeJS: 'list2', 3, { size: 3,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _origin: 0,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _capacity: 3,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _level: 5,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _root: null,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   _tail: { array: [ 1, [], 3 ], ownerID: {} },
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __ownerID: undefined,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __hash: undefined,
01-14 10:22:26.568  4347  4982 I ReactNativeJS:   __altered: false }
```

So basically it seems that the element is replaced by empty array instead of being removed from collection. It only happens when running without debugger attached, so I guess it may be dependent on JS engine.

**Comment by lpatalas** [comment]
After more digging I found that the culprit was missing `Symbol.iterator` for array on Android. It works after loading polyfill.

**Comment by sdeleon28** [comment]
I hadn't seen this had been reported and created a test case to reproduce it. It's here for future reference -> https://github.com/sdeleon28/RNAndroidImmutableBug

**Comment by sdeleon28** [comment]
Potential quickfix for future lost travelers https://github.com/facebook/react-native/issues/4676#issuecomment-340290485

**Comment by Ilya33** [comment]
Still reproduce on react-native 0.58.6

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/fad903a6b04feeeac91a6f1b8acf486140eff7f2

**Comment by jdeniau** [comment]
Hi!
The 4.0.0-rc.14 version has been released and should fix this issue.
Feel free to test it and re-open this issue if the issue is still present.
Thank you!
